src/library/dlgcoverartfullsize.cpp

```diff
@@ -1,42 +1,193 @@
+        disconnect(m_pLoadedTrack.get(), SIGNAL(coverArtUpdated()),
+                   this, SLOT(slotTrackCoverArtUpdated()));
     }
+    m_pLoadedTrack = pTrack;
+    if (m_pLoadedTrack != nullptr) {
+        QString windowTitle;
+        const QString albumArtist = m_pLoadedTrack->getAlbumArtist();
+        const QString artist = m_pLoadedTrack->getArtist();
+        const QString album = m_pLoadedTrack->getAlbum();
+        const QString year = m_pLoadedTrack->getYear();
+        if (!albumArtist.isEmpty()) {
+            windowTitle = albumArtist;
+        } else if (!artist.isEmpty()) {
+            windowTitle += artist;
+        }
+        if (!album.isEmpty()) {
+            if (!windowTitle.isEmpty()) {
+                windowTitle += " - ";
+            }
+            windowTitle += album;
+        }
+        if (!year.isEmpty()) {
+            if (!windowTitle.isEmpty()) {
+                windowTitle += " ";
+            }
+            windowTitle += QString("(%1)").arg(year);
+        }
+        setWindowTitle(windowTitle);
 
-    // If cover is bigger than Mixxx, it must be resized!
-    // In this case, it need to do a small adjust to make
-    // this dlg a bit smaller than the Mixxx window.
-    QSize mixxxSize = largestWindowSize / qreal(1.2);
-    if (pixmap.height() > mixxxSize.height()
-            || pixmap.width() > mixxxSize.width()) {
-        pixmap = pixmap.scaled(
-            mixxxSize, Qt::KeepAspectRatio, Qt::SmoothTransformation);
+        connect(m_pLoadedTrack.get(), SIGNAL(coverArtUpdated()),
+                this, SLOT(slotTrackCoverArtUpdated()));
     }
-    resize(pixmap.size());
-    coverArt->setPixmap(pixmap);
 
-    show();
-    move(QApplication::desktop()->screenGeometry().center() - rect().center());
-    raise();
-    activateWindow();
+    slotTrackCoverArtUpdated();
+}
+
+void DlgCoverArtFullSize::slotTrackCoverArtUpdated() {
+    if (m_pLoadedTrack != nullptr) {
+        CoverArtCache::requestCover(*m_pLoadedTrack, this);
+    }
+}
+
+void DlgCoverArtFullSize::slotCoverFound(const QObject* pRequestor,
+                                         const CoverInfo& info, QPixmap pixmap,
+                                         bool fromCache) {
+    Q_UNUSED(info);
+    Q_UNUSED(fromCache);
+
+    if (pRequestor == this && m_pLoadedTrack != nullptr &&
+            m_pLoadedTrack->getCoverHash() == info.hash) {
+        // qDebug() << "DlgCoverArtFullSize::slotCoverFound" << pRequestor << info
+        //          << pixmap.size();
+        m_pixmap = pixmap;
+        move(QApplication::desktop()->screenGeometry().center() - rect().center());
+        if (m_pixmap.isNull()) {
+            close();
+        } else {
+            // Scale down dialog to screen size if the pixmap is larger than the screen
+            QSize dialogSize = m_pixmap.size();
+            const QSize availableScreenSpace =
+                QApplication::desktop()->availableGeometry().size();
```

Comment:
This is too large in full screen mode. The cover art window extends below the bottom of the screen, because it will always be placed underneath the menu bar of the main window and is not able to move above it. This restriction does not apply if the application uses a regular window. @Be-ing Any ideas for an easy fix?

Reply:
What window manager do you use? With KWin (X11), I have a different issue with full screen mode. In that case, the cover art window behaves as if my bottom menu bar is still showing as if I was not in full screen mode.

Reply:
We seem to have different issues with different software setups and these very well could change with Qt5, so I am doubtful it is worthwhile to perfect all the edge cases right now.

Reply:
Using availableGeometry().size() * 0.9 instead of the desktop size keeps the main window visible in around the cover art popup in all directions and works much better for me. Simple but effective.

Reply:
Clever workaround!